test/hooks/useResult.ts

```diff
@@ -21,6 +23,19 @@ describe("useResult", () => {
     expect(Result.isSuccess(result.current.result)).toBe(true)
   })
 
+  it("override Provider value", async () => {
+    const [Override] = RuntimeContext.overrideLayer(context, Layer.succeed(foo, { value: 2 }))
```

Comment:
This pattern feels "unnatural" in react settings. We need to think if we want to do this or something else..